runtime/src/qio/sys.c

```diff
@@ -97,12 +97,73 @@
 // Should be available in sys_xsi_strerror_r.c
 extern int sys_xsi_strerror_r(int errnum, char* buf, size_t buflen);
 
-void sys_init_sys_sockaddr(sys_sockaddr_t* addr)
+void sys_init_sys_sockaddr_t(sys_sockaddr_t* addr)
 {
   memset(addr, 0, sizeof(sys_sockaddr_t));
   addr->len = sizeof(sys_sockaddr_storage_t);
 }
 
+int sys_getsockaddr_family(sys_sockaddr_t* addr){
+  return addr->addr.ss_family;
+}
+
+int sys_fill_sys_sockaddr_t(sys_sockaddr_t* addr, const char* host, u_int16_t port, int family)
```

Comment:
just judging from this function signature I would expect it to work with DNS lookup in the event host is a string like www.hpe.com. Maybe we can make it clear that this only works with IP address strings in the documentation and variable names.

Reply:
sure